Multiple Updates (draft)

IceOfWraith opened the pull request:
This is the beginning of my overhaul of the config.getamp.sh site. Nothing crazy so far, but more is being added as I figure out the code:

-Placeholder is set to DefaultValue
-Suffix added
-URI adjusted to new port config and to use `:` instead of `/`
-SteamAppID corrected to Client App ID instead of Server
-Many formatting changes due to VS Studio cleaning up the code layout
-Can generate Wine & Proton configs with default of xvfb
-Corrected mismatched HTML tags
-Added section breaks
-Added New Ports Mapping
-Added Meta Config Manifest
-Added all kvp default values
-Enabled multiple update stages with all update source types included
-Keywords populate from Display Name
-ENUM VALUES!!!
-Surprise! No more regex! Evolved Mike's base function from AMP for creating regex to design a user-friendly approach

To do:
-Verify all settings
-Finish adding kvp keys
-Error checking logic
-Bug fixes
-Clean up
-User testing

IceOfWraith committed:
Multiple Updates (draft)

This is the beginning of my overhaul of the config.getamp.sh site. Nothing crazy so far, but more is being added as I figure out the code:

-Placeholder is set to DefaultValue
-Suffix added
-URI adjusted to new port config and to use `:` instead of `/`
-SteamAppID corrected to Client App ID instead of Server

IceOfWraith committed:
Add Port Settings

IceOfWraith committed:
Proton and Wine support (not complete)

IceOfWraith committed:
Fixed HTML errors & warninig, Proton works, Wine... does not

IceOfWraith committed:
Switch compatibility to radio buttons

IceOfWraith committed:
Begin updates to new port system (Saving progress)

IceOfWraith committed:
Fix my somehow working but wrong code

IceOfWraith committed:
Add Placeholder setting values and add setting options

IceOfWraith committed:
Bump version, further port settings, config settings additions

IceOfWraith committed:
Working on ports

IceOfWraith committed:
Port Description

IceOfWraith committed:
Ports, ports, ports

IceOfWraith committed:
Add all default settings

IceOfWraith committed:
Does anyone even read these?

IceOfWraith committed:
Add Meta Config Manifest

IceOfWraith committed:
Clean up layout

IceOfWraith committed:
Add Main Game Port

IceOfWraith committed:
CLI Args

IceOfWraith committed:
2 just because

IceOfWraith committed:
Links should open in new tabs

IceOfWraith committed:
Begin implementing multiple update stages

IceOfWraith committed:
More progress towards update stages

IceOfWraith committed:
Finish most update settings + Dark Mode

IceOfWraith committed:
Saving work on EnumValues

IceOfWraith committed:
More enum value changes

IceOfWraith committed:
Create index.html

IceOfWraith committed:
Delete index.html

IceOfWraith committed:
Rename default.html to index.html

index.html is required by GitHub Pages

gOOvER commented:
i hope, you are still working on i, because i would love to see the changes in the config generator :)

IceOfWraith commented:
Oh very much so! I took a break to catch up on my huge backlog. I'm mostly caught up and plan to shift gears back to this. I was also stuck on a part for too long and needed to take a break to clear my thoughts on how to do it. Haha

IceOfWraith commented:
Here's a preview of it. Not fully functioning though so don't get too deep into it yet!

https://config.iceofwraith.com

IceOfWraith committed:
Update generator.js

IceOfWraith committed:
Update index.html

IceOfWraith committed:
fdasiobjveroa

IceOfWraith committed:
Update generator.js

IceOfWraith committed:
Update generator.js

IceOfWraith committed:
Update generator.js

IceOfWraith committed:
Credit where credit is due

IceOfWraith committed:
Regex? More like REJECT! Am I right?

IceOfWraith committed:
ForceDownloadPlatform was missing

IceOfWraith committed:
Match kvp settings up again

IceOfWraith committed:
More

IceOfWraith committed:
Logic for Base/Working Directories

IceOfWraith committed:
Some layout and code cleanup

IceOfWraith commented:
I've been keeping the original comment updated with status and you can click the link above to use the pre-release version. I think it's usable as is. But it probably has some bugs.

Justman10000 commented:
Oh, darkmode and multistages... 🙂

IceOfWraith committed:
Combined download zip and manifest.json creation

IceOfWraith committed:
Update generator.js

IceOfWraith committed:
Update instructions for using the config

IceOfWraith committed:
Missing line separator

IceOfWraith committed:
Fix Steam Client ID

IceOfWraith committed:
Revert "Fix Steam Client ID"

This reverts commit 983ceac2966d3b59b3360c9ec3d8dcd894609fd1.

IceOfWraith committed:
Fix Steam Client ID

IceOfWraith committed:
Update generator.js

IceOfWraith committed:
Multiple Changes

Clean up some code and prevent modals from closing from clicking the background.

IceOfWraith committed:
Rearrange Update Stages

This rearranges the update stages to fix a bug where AMP refused the order they were in.

IceOfWraith committed:
Restrict to 1 of each port type except Custom

IceOfWraith committed:
Compatibility and Display Image Source

IceOfWraith committed:
Update index.html

IceOfWraith committed:
Hide values when no Linux

IceOfWraith committed:
Remove Startup Mode from GUI - Too much info

IceOfWraith committed:
Lots of code formatting cleanup and minor tweaks

IceOfWraith committed:
Oops

IceOfWraith committed:
Update index.html

IceOfWraith committed:
Code cleanup

IceOfWraith committed:
Fix ports

IceOfWraith committed:
css cleanup

IceOfWraith committed:
Fixes

IceOfWraith committed:
Error checking

IceOfWraith committed:
Revert "css cleanup"

This reverts commit af7b3a93d449eaea3384f1d5f651513db3525c0b.

IceOfWraith committed:
Revert "Fixes"

This reverts commit a3fcafd351f2825b90043e205317c619b4915309.

IceOfWraith committed:
RRRRaS

IceOfWraith committed:
Rewind to the working code

IceOfWraith committed:
Fixing deployments

IceOfWraith committed:
Fixing still

Justman10000 commented:
New preview [here](https://gcg.justman10000.de)!

IceOfWraith commented:
I've already got it hosted here:

https://config.iceofwraith.com

Justman10000 commented:
> I've already got it hosted here:
> 
> https://config.iceofwraith.com

😢😢😢

IceOfWraith commented:
I had to set it up overnight because GitHub Pages decided to stop deploying. I was trying to revert multiple commits that I didn't realize were pushed to the public side when I was playing around wildly with the code. Lol

IceOfWraith committed:
favicon

IceOfWraith committed:
Minor changes

IceOfWraith committed:
Update generator.js

IceOfWraith committed:
Fix Update Stages

IceOfWraith committed:
Update Author

IceOfWraith committed:
Update generator.js

IceOfWraith committed:
Fix Issues and Add KO Validate

IceOfWraith committed:
Add AppConfigId and filter Errors and isValie

IceOfWraith committed:
Fix compatibility validation on Linux
